core/templates/components/state-directives/outcome-editor/outcome-destination-editor.component.ts

```diff
@@ -16,186 +16,208 @@
  * @fileoverview Component for the outcome destination editor.
  */
 
-require('components/graph-services/graph-layout.service.ts');
-require('domain/utilities/url-interpolation.service.ts');
-require(
-  'pages/exploration-editor-page/services/editor-first-time-events.service.ts');
-require(
-  'components/state-editor/state-editor-properties-services/' +
-  'state-editor.service.ts');
-require('services/editability.service.ts');
-require('services/user.service.ts');
-require('services/stateful/focus-manager.service.ts');
-
-require('constants.ts');
-
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+import { downgradeComponent } from '@angular/upgrade/static';
 import { Subscription } from 'rxjs';
-
-angular.module('oppia').component('outcomeDestinationEditor', {
-  bindings: {
-    outcomeHasFeedback: '=',
-    outcome: '=',
-    addState: '='
-  },
-  template: require('./outcome-destination-editor.component.html'),
-  controllerAs: '$ctrl',
-  controller: [
-    '$rootScope', '$timeout', 'EditorFirstTimeEventsService',
-    'FocusManagerService', 'StateEditorService', 'StateGraphLayoutService',
-    'UserService', 'ENABLE_PREREQUISITE_SKILLS',
-    'EXPLORATION_AND_SKILL_ID_PATTERN', 'MAX_STATE_NAME_LENGTH',
-    'PLACEHOLDER_OUTCOME_DEST',
-    function(
-        $rootScope, $timeout, EditorFirstTimeEventsService,
-        FocusManagerService, StateEditorService, StateGraphLayoutService,
-        UserService, ENABLE_PREREQUISITE_SKILLS,
-        EXPLORATION_AND_SKILL_ID_PATTERN, MAX_STATE_NAME_LENGTH,
-        PLACEHOLDER_OUTCOME_DEST) {
-      var ctrl = this;
-      ctrl.directiveSubscriptions = new Subscription();
-      var currentStateName = null;
-      ctrl.isSelfLoop = function() {
-        return ctrl.outcome.dest === currentStateName;
-      };
-
-      ctrl.onDestSelectorChange = function() {
-        if (ctrl.outcome.dest === PLACEHOLDER_OUTCOME_DEST) {
-          FocusManagerService.setFocus('newStateNameInputField');
+import cloneDeep from 'lodash/cloneDeep';
+import { StateGraphLayoutService } from 'components/graph-services/graph-layout.service';
+import { StateEditorService } from 'components/state-editor/state-editor-properties-services/state-editor.service';
+import { EditorFirstTimeEventsService } from 'pages/exploration-editor-page/services/editor-first-time-events.service';
+import { FocusManagerService } from 'services/stateful/focus-manager.service';
+import { UserService } from 'services/user.service';
+import { AppConstants } from 'app.constants';
+import { Outcome } from 'domain/exploration/OutcomeObjectFactory';
+
+interface DestChoices {
```

Comment:
Maybe use a more expressive name here for the interface. `DestinationChoice` seems better as it is more expressive and also drops the `s` which tell that this object is a single choice